components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -305,7 +310,21 @@ public OAuth2AccessTokenRespDTO issue(OAuth2AccessTokenReqDTO tokenReqDTO)
 
         if (tokReqMsgCtx.getScope() != null && OAuth2Util.isOIDCAuthzRequest(tokReqMsgCtx.getScope())) {
             IDTokenBuilder builder = OAuthServerConfiguration.getInstance().getOpenIDConnectIDTokenBuilder();
-            tokenRespDTO.setIDToken(builder.buildIDToken(tokReqMsgCtx, tokenRespDTO));
+            String idToken = builder.buildIDToken(tokReqMsgCtx, tokenRespDTO);
+            try {
+                JWT jwt = JWTParser.parse(idToken);
```

Comment:
The DefaultIDTokenBuilder#buildIDToken method actually creates the JWT token object, and then serialize it. [1] In the above code, we take that serialized string and parse it again into a JWT token object. 

IMO, this is not necessary and will introduce additional latency.

[1] https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/6855d9a8bd26c74b2cdd6bdf02ccd2be7ba75eea/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilder.java#L316